src/js/components/receiving/PartialReceivingPage.jsx

```diff
@@ -735,33 +735,134 @@ class PartialReceivingPage extends Component {
     }));
   }
 
+  static mapShipmentItems(shipmentItems, rowIndex, editedLines) {
+    // Group items for: before edited row, the edited row's original item, items after edited row
+    // To be able to append a new item just underneath (or replace) the original item
+    const shipmentItemsGrouped = shipmentItems.reduce((acc, item, idx) => {
+      if (idx < rowIndex) {
+        return {
+          ...acc,
+          itemsBeforeCurrentRow: [...acc.itemsBeforeCurrentRow, item],
+        };
+      }
+      if (idx === rowIndex) {
+        return {
+          ...acc,
+          originalItem: item,
+        };
+      }
+      return {
+        ...acc,
+        itemsAfterCurrentRow: [...acc.itemsAfterCurrentRow, item],
+      };
+    }, { itemsBeforeCurrentRow: [], originalItem: null, itemsAfterCurrentRow: [] });
```

Comment:
I think you can simplify that code using `_.groupBy` from lodash

Reply:
I don't know how I would be able to use the `_.groupBy` from lodash here - from what I can see in the [docs](https://lodash.com/docs/4.17.15#groupBy), I don't find it useful in that specific case, where I base the "key" on whether the index is lower, equal or higher than x.
I believe that this `.reduce` is quite readable, isn't it?